File: packages/app/src/services/software/definitions/installForge.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'installforge';
	public name = 'InstallForge';
	public category = [SoftwareCategory.Development];
	public downloadName = 'IFSetup.exe';
	public requiresAdmin = true;
	public icon = 'installforge.png';
	public homepage = 'https://installforge.net';

	public async resolveDownloadUrl() {
		return ok('https://installforge.net/downloads/?i=IFSetup');
	}
}

File: src/lib/software/installforge.py
from src.lib.software import BaseSoftware

class InstallForge(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'IFSetup.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://installforge.net/downloads/?i=IFSetup')
